modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/Archetyper.java

```diff
@@ -167,6 +214,89 @@ public ArchetypeGenerationResult generateProject(
 		return archetypeGenerationResult;
 	}
 
+	private static File _getArchetypeFile(String artifactId, File file)
+		throws IOException {
+
+		try (JarFile jarFile = new JarFile(file)) {
+			Enumeration<JarEntry> enumeration = jarFile.entries();
+
+			while (enumeration.hasMoreElements()) {
+				JarEntry jarEntry = enumeration.nextElement();
+
+				if (jarEntry.isDirectory()) {
+					continue;
+				}
+
+				String name = jarEntry.getName();
+
+				if (!name.startsWith(artifactId + "-")) {
+					continue;
+				}
+
+				Path archetypePath = Files.createTempFile(
+					"temp-archetype", null);
+
+				Files.copy(
+					jarFile.getInputStream(jarEntry), archetypePath,
+					StandardCopyOption.REPLACE_EXISTING);
+
+				File archetypeFile = archetypePath.toFile();
+
+				archetypeFile.deleteOnExit();
+
+				return archetypeFile;
+			}
+		}
+
+		return null;
+	}
+
+	private static File _getTemplateFile(
+			ProjectTemplatesArgs projectTemplatesArgs)
+		throws Exception {
+
+		String template = projectTemplatesArgs.getTemplate();
+
+		for (File archetypesDir : projectTemplatesArgs.getArchetypesDirs()) {
+			if (archetypesDir.isDirectory()) {
+				try (DirectoryStream<Path> directoryStream =
+						Files.newDirectoryStream(
+							archetypesDir.toPath(), "*.project.templates.*")) {
+
+					Iterator<Path> iterator = directoryStream.iterator();
```

Comment:
Instead of the iterator, you can you a `for` loop: `for (Path path : directoryStream)`.